pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/PlaybackSessionManager.kt

```diff
@@ -0,0 +1,209 @@
+         */
+        fun onSessionCreated(session: Session)
+
+        /**
+         * On session finished
+         *
+         * @param session
+         */
+        fun onSessionFinished(session: Session)
+
+        /**
+         * On current session
+         *
+         * @param session
+         */
+        fun onCurrentSession(session: Session)
+    }
+
+    /**
+     * Session
+     *
+     * @property mediaItem The [MediaItem] linked to the session.
+     */
+    class Session(val mediaItem: MediaItem) {
+        /**
+         * Unique Session Id
+         */
+        val sessionId = UUID.randomUUID().toString()
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as Session
+
+            return sessionId == other.sessionId
+        }
+
+        override fun hashCode(): Int {
+            return sessionId.hashCode()
+        }
+    }
+
+    private val sessions = HashMap<String, Session>()
+
+    /**
+     * Listener
+     */
+    var listener: Listener? = null
+
+    /**
+     * Current session
+     */
+    var currentSession: Session? = null
+        private set(value) {
+            if (field != value) {
+                field?.let {
+                    listener?.onSessionFinished(it)
+                    sessions.remove(it.sessionId)
+                }
+                field = value
+                field?.let {
+                    listener?.onCurrentSession(it)
+                }
+            }
+        }
+
+    /**
+     * Get or create session from a [MediaItem]
+     *
+     * @param mediaItem The [MediaItem].
+     * @return a [Session] associated with mediaItem.
+     */
+    fun getOrCreateSession(mediaItem: MediaItem): Session {
+        val session = sessions.firstNotNullOfOrNull { if (it.value.mediaItem.isTheSame(mediaItem)) it.value else null }
+        if (session == null) {
+            val newSession = Session(mediaItem)
+            sessions[newSession.sessionId] = newSession
+            listener?.onSessionCreated(newSession)
+            if (currentSession == null) {
+                currentSession = newSession
+            }
+            return newSession
+        }
+        return session
+    }
+
+    override fun onPositionDiscontinuity(
+        eventTime: EventTime,
+        oldPosition: Player.PositionInfo,
+        newPosition: Player.PositionInfo,
+        reason: Int
+    ) {
+        super.onPositionDiscontinuity(eventTime, oldPosition, newPosition, reason)
+        val oldItemIndex = oldPosition.mediaItemIndex
+        val newItemIndex = newPosition.mediaItemIndex
+        Log.d(TAG, "onPositionDiscontinuity reason = ${StringUtil.discontinuityReasonString(reason)}")
+        if (oldItemIndex != newItemIndex && !eventTime.timeline.isEmpty) {
+            val newSession = getOrCreateSession(eventTime.timeline.getWindow(newItemIndex, Timeline.Window()).mediaItem)
```

Comment:
Shouldn't `Timeline.Window()` be extracted in a field, so we don't recreate it every time?